collect_app/src/main/java/org/odk/collect/android/receivers/NetworkReceiver.java

```diff
@@ -58,28 +60,51 @@ public void onReceive(Context context, Intent intent) {
         String action = intent.getAction();
         ConnectivityManager manager = (ConnectivityManager) context.getSystemService(
                 Context.CONNECTIVITY_SERVICE);
+
         NetworkInfo currentNetworkInfo = manager.getActiveNetworkInfo();
 
-        if (action.equals(ConnectivityManager.CONNECTIVITY_ACTION)) {
+        if (action.equals(ConnectivityManager.CONNECTIVITY_ACTION)
+                || action.equals("org.odk.collect.android.FormSaved")) {
             if (currentNetworkInfo != null
                     && currentNetworkInfo.getState() == NetworkInfo.State.CONNECTED) {
```

Comment:
`currentNetworkInfo.isConected()` would be better here.

Reply:
I men out of https://github.com/opendatakit/collect/blob/b01d097e3efbbc13e852663cbb839d5e1df4ce84/collect_app/src/main/java/org/odk/collect/android/receivers/NetworkReceiver.java#L66